packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @fileoverview Hint to validate if the doctype is correct
+ */
+
+import { Category } from 'hint/dist/src/lib/enums/category';
+import { HintScope } from 'hint/dist/src/lib/enums/hintscope';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, HintMetadata, FetchEnd, ProblemLocation } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+import { MatchInformation } from './types';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class implements IHint {
+
+    public static readonly meta: HintMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `This hint checks if the HTML is using the most modern DOCTYPE.`
+        },
+        id: 'doctype',
+        schema: [],
+        scope: HintScope.any
+    }
+
+    public constructor(context: HintContext) {
+        const correctLine = 0;
+        const doctypeRegExp = new RegExp(`(<!doctype\\s+(html)\\s*>)(.+)?`, 'gi');
+
+        const defaultProblemLocation: ProblemLocation = {
+            column: 0,
+            line: 0
+        };
+
+        const report = async (resource: string, message: string, problemLocation = defaultProblemLocation): Promise<void> => {
+            await context.report(resource, null, message, undefined, problemLocation);
+        };
+
+        const getCurrentDoctypeProblemLocation = (text: string): ProblemLocation[] => {
+            const lines = text.split('\n');
```

Comment:
Should we use [`os.EOL`](https://nodejs.org/api/os.html#os_os_eol)? 

Or, am I missing something and we are normalizing things at some point?

Reply:
Thanks for the feedback, because of this bug with template strings https://github.com/nodejs/node/issues/3121 we had to normalize the content, but only in the tests, which use template strings.

We were also looking for examples using os.EOL in the code and found instead that '/n' was used a lot instead, so maybe that should be looked into.

Reply:
Ok. 🤷🏻‍♂️

Reply:
![I find your lack of faith... disturbing](http://www.quirkbooks.com/sites/default/files/u84/choke.gif)

Reply:
@molant the feedback was applied yesterday, do you want to reset it?